### PR TITLE
RankWidget in a dialog

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### What has been done to verify that this works as intended?
I've tested this change using the attached form.

#### Why is this the best possible solution? Were any other approaches considered?
It's an improvement required by users on the forum to make this widget consistent with others that have buttons like DateWidget.

#### Are there any risks to merging this code? If so, what are they?
Can't see any risk here.

#### Do we need any specific form for testing your changes? If so, please attach one.
I used:
[rank.xml.txt](https://github.com/opendatakit/collect/files/2197575/rank.xml.txt)

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew pmd checkstyle lintDebug spotbugsDebug` and confirmed all checks still pass.
- [ ] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

**Comment by lognaturel** [comment]
I'll admit that it's not terrible. The dialog does give us more space instead of less. ~One advantage I hadn't considered is that we may not need to use a long press since it's not possible to swipe forward on a dialog anyway, right?~ Nevermind -- we may need to scroll the dialog if we have a long list.

If this is the direction we decide to go in, I would just ask that the answer summary under the button show the values as a numbered list something like:
```
1. Cherimoya
2. Blueberry
3. Apple
```

**Comment by yanokwa** [comment]
I prefer this dialog version over the current version and I'd like get it into the code freeze. @grzesiek2010 can you make the improvements to the code and add @lognaturel's suggestion?

**Comment by grzesiek2010** [comment]
It's ready

File: collect_app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -613,6 +613,6 @@
     <string name="hide_old_form_versions_setting_title">Hide old form versions</string>
     <string name="hide_old_form_versions_setting_summary">Only the newest version will appear in Fill Blank Form</string>
     <string name="seconds">Seconds</string>
-    <string name="start_ranking">Start ranking</string>
+    <string name="sort_out_items">Sort out items</string>
```
Comment (lognaturel) [inline collect_app/src/main/res/values/strings.xml]: This text doesn't make sense to me. @grzesiek2010 what did you not like about "Start ranking"? Perhaps "Rank items"? @yanokwa?

Comment (grzesiek2010) [reply collect_app/src/main/res/values/strings.xml]: `Start ranking` is ok if there is no answer yet.  I didn't like it was always the same title. 
`Rank items` would be ok.

Comment (lognaturel) [reply collect_app/src/main/res/values/strings.xml]: Make sense! Let's go with `Rank items`.

Comment (grzesiek2010) [reply collect_app/src/main/res/values/strings.xml]: Done.

Comment (lognaturel) [reply collect_app/src/main/res/values/strings.xml]: 🙏

**Comment by mmarciniak90** [comment]
@grzesiek2010 app crashes when user minimize or lock the screen if RankWidget dialog is opened.

**Comment by grzesiek2010** [comment]
@mmarciniak90 ok I've just fixed the problem.

**Comment by mmarciniak90** [comment]
@grzesiek2010 thanks for so quick fix :+1:  

**Tested with success**
Verified on Android: 4.1, 4.4, 5.1, 6.0

|screenshot_20180717-150517|screenshot_20180717-150614|
|---|---|

@opendatakit-bot label "behavior verified"
@opendatakit-bot unlabel "needs testing"

**Review by lognaturel — APPROVED** [review]

**Pull request merged**